Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,282 @@
+  have nhd : H.carrierᶜ ∈ nhds σ := H.isClosed'.isOpen_compl.mem_nhds h
+  rw [GroupFilterBasis.nhds_eq (x₀ := σ) (galGroupBasis k K)] at nhd
+  rcases nhd with ⟨b,⟨gp,⟨L,hL,eq'⟩,eq⟩,sub⟩
+  rw [← eq'] at eq
+  have sub : σ • b ⊆ H.carrierᶜ := Set.smul_set_subset_iff.mpr sub
+  have := hL.out
+  let L' : FiniteGaloisIntermediateField k K := {
+    normalClosure k L K with
+    finiteDimensional := normalClosure.is_finiteDimensional k L K
+    isGalois := IsGalois.normalClosure k L K }
+  have compl : σ • L'.1.fixingSubgroup.carrier ⊆ H.carrierᶜ :=
+    fun ⦃a⦄ d ↦ sub ((Set.set_smul_subset_set_smul_iff.mpr <| eq ▸ (fun σ h =>
+    (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mpr fun y hy => (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mp
+    h y (IntermediateField.le_normalClosure L hy))) d)
+  have fix : ∀ x ∈ IntermediateField.fixedField H.toSubgroup ⊓ ↑L', σ x = x :=
+    fun x hx ↦ ((mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mp hσ) x hx.1
+  rw [restrict_fixedField H.1 L'.1] at fix
+  have : (restrictNormalHom L') σ ∈ (Subgroup.map (restrictNormalHom L') H.1) := by
+    rw [← IntermediateField.fixingSubgroup_fixedField (Subgroup.map (restrictNormalHom L') H.1)]
+    apply (mem_fixingSubgroup_iff (L' ≃ₐ[k] L')).mpr
+    intro y hy
+    apply Subtype.val_injective
+    simp only [AlgEquiv.smul_def, restrictNormalHom_apply L'.1 σ y,
+      fix y.1 ((IntermediateField.mem_lift y).mpr hy)]
+  rcases this with ⟨h, mem, eq⟩
+  have : h ∈ σ • L'.1.fixingSubgroup.carrier := by
+    use σ⁻¹ * h
+    simp only [Subsemigroup.mem_carrier, Submonoid.mem_toSubsemigroup, Subgroup.mem_toSubmonoid,
+      smul_eq_mul, mul_inv_cancel_left, and_true]
+    apply (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mpr
+    intro y hy
+    simp only [AlgEquiv.smul_def, AlgEquiv.mul_apply]
+    have : ((restrictNormalHom L') h ⟨y,hy⟩).1 = ((restrictNormalHom L') σ ⟨y,hy⟩).1 := by rw [eq]
+    rw [restrictNormalHom_apply L'.1 h ⟨y, hy⟩, restrictNormalHom_apply L'.1 σ ⟨y, hy⟩] at this
+    simp only [this, ← AlgEquiv.mul_apply, inv_mul_cancel, one_apply]
+  absurd compl
+  apply Set.not_subset.mpr
+  use h
+  simpa only [this, Set.mem_compl_iff, Subsemigroup.mem_carrier, Submonoid.mem_toSubsemigroup,
+    Subgroup.mem_toSubmonoid, not_not, true_and] using mem
+
+/-- The Galois correspondence from intermediate fields to ClosedSubgroup. -/
+def IntermediateFieldEquivClosedSubgroup [IsGalois k K] :
+    IntermediateField k K ≃o (ClosedSubgroup (K ≃ₐ[k] K))ᵒᵈ where
+  toFun := fun L =>
+    { L.fixingSubgroup with
+      isClosed' := fixingSubgroup_isClosed L }
+  invFun := fun H => IntermediateField.fixedField H.1
+  left_inv := fun L => fixedField_fixingSubgroup L
+  right_inv := by
+    intro H
+    simp_rw [fixingSubgroup_fixedField H]
+    rfl
+  map_rel_iff' := by
+    intro L₁ L₂
+    show L₁.fixingSubgroup ≥ L₂.fixingSubgroup ↔ L₁ ≤ L₂
+    rw [← fixedField_fixingSubgroup L₂, IntermediateField.le_iff_le, fixedField_fixingSubgroup L₂]
+
+/-- The Galois correspondence as a `GaloisInsertion` -/
```

Comment:
This is an equivalence right? What is the advantage of making it a Galois insertion? Surely the correct Galois insertion (or possibly coinsertion) is between intermediate fields and *all* subgroups (not just closed ones)? Because this gives you new information. It's an insertion (or coinsertion) because if you start with the field, then make the group, then make the field, it's the field you started with. The same isn't true for the groups of course, you end up with the closure of the group.

I'm slightly surprised that you can't just define the Galois insertion here in one line, invoking some result which says "an order isomorphism gives a Galois insertion". [looks in docs] yeah how about [OrderIso.toGaloisInsertion](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/GaloisConnection.html#OrderIso.toGaloisInsertion) ? I'm not clear what this is giving you.

Reply:
I just copied these two results from the original `Galois` file, sorry that I doesn't know the reason clearly.

Reply:
The one-line proof works well for Insertion, it is the `coinsertion` that can be changed into subgroup.

Reply:
Thanks! Yes, this is what I was aiming for -- it seems like this has more content. Thanks! Perhaps if you take `toDual` on the other factor then it becomes an insertion? 

You can probably even do some kind of insertion/coinsertion from Set K to Set (K ≃ₐ[k] K) with the same definition (of course the image of the maps are still intermediate fields and closed subgroup, I'm just observing that the maps themselves can be defined on this much bigger domain), although I don't know if these things are useful.